subtrack.Tests/SubscriptionCalculatorTests/GetNextPaymentDateTests.cs

```diff
@@ -39,6 +39,46 @@ public void GetNextPaymentDate_WhenLastPaymentIsLastDayOfMonth_ReturnsPaymentDat
             // Act
             var result = _sut.GetNextPaymentDate(subscription);
 
+            // Assert
+            Assert.Equal(expected, result);
+        }
+        [Fact]
+        public void GetNextPaymentDate_WhenLastPaymentIsLastDayOfFebruary_ReturnsPaymentDateWithLastDayOfNextMonth()
+        {
+            // Arrange
+            var lastPayment = new DateTime(2023, 02, 28);
+            var subscription = new Subscription { LastPayment = lastPayment, FirstPaymentDay = lastPayment.Day };
+            var expected = new DateTime(2023, 03, 31);
+
+            // Act
+            var result = _sut.GetNextPaymentDate(subscription);
+
+            // Assert
+            Assert.Equal(expected, result);
+        }
+        [Fact]
+        public void GetNextPaymentDate_WhenLastPaymentIsLastDayOfJanuary_ReturnsPaymentDateWithLastDayOfNextMonth()
+        {
+            // Arrange
+            var subscription = new Subscription { LastPayment = new DateTime(2023, 01, 31) };
```

Comment:
Theese tests are very similar, i would recommend using [Theory]. Make sure to also set the FirstPaymentDay